src/test/java/org/osiam/client/OsiamUserServiceTest.java

```diff
@@ -105,6 +104,52 @@ public void invalid_filter_generates_bad_request() {
         service.searchUsers(new QueryBuilder().filter(filter).build(), accessToken);
     }
 
+    @Test
+    public void passing_attributes_to_a_request_for_all_users_generates_the_correct_request() {
```

Comment:
This is indeed confusing. I first thought "Codacy is right". But after studying the test for a minute, Codacy seems to be wrong. If I understand the code correctly, you're verifying that the connector generates the correct HTTP request, because otherwise Mock Server won't reply with a SCIM list response. The issue is that it takes a while to understand the test and why it asserts something after all. I think part of this issue is, that the "assertions" are defined at the top of test and, worse yet, are contained in method called `when(…)`. Do you think that Mock Server's API allows us to make the assertions clearer?